github/pulls.go

```diff
@@ -250,17 +250,21 @@ type PullRequestMergeResult struct {
 
 type pullRequestMergeRequest struct {
 	CommitMessage *string `json:"commit_message"`
+	Squash        *bool   `json:"squash,omitempty"`
 }
 
 // Merge a pull request (Merge Button™).
 //
 // GitHub API docs: https://developer.github.com/v3/pulls/#merge-a-pull-request-merge-buttontrade
-func (s *PullRequestsService) Merge(owner string, repo string, number int, commitMessage string) (*PullRequestMergeResult, *Response, error) {
+func (s *PullRequestsService) Merge(owner string, repo string, number int, commitMessage string, squash bool) (*PullRequestMergeResult, *Response, error) {
```

Comment:
probably put this into an Options struct, so we don't have to make another breaking change if they add more options in the future.